File: Prácticas/En curso/Pokedex2.0/PokedexAPI/PokeType.cs
using System;

namespace PokedexAPI
{
    public class PokeType
    {
        public int id { get; }
        public string name { get; set; }
        public RGBColor Color { get; set; }

        public class RGBColor
        {
            private int r { get; }
            private int g { get; }
            private int b { get; }
            public RGBColor(int r, int g, int b)
            {
                this.r = r;
                this.g = g;
                this.b = b;
            }
        }
    }
}

File: practices/completed/full-projects/PokedexDesktop/src/models/PokeType.java
package models;

import java.awt.Color;

/**
 * Method that simulate a Pokemon Type. It has a color attached, a name and a id.
 * 
 * @author AdriGB
 *
 */
public class PokeType implements Comparable<PokeType> {
	
	// Attributes
	private int idT;
	private String name;
	private Color color;
	
	/**
	 * Constructor that receives main attributes.
	 * 
	 * @param idT
	 * @param nombre
	 * @param color This color will define how this type will be shown in this application
	 */
	public PokeType (int idT, String nombre, Color color) {
		this.idT = idT;
		this.name = nombre;
		this.color = color;
	}
	
	/** GETTERS AND SETTERS */
	public Color getColor() {
		return color;
	}

	public int getIdT() {
		return idT;
	}

	public String getName() {
		return name;
	}

	@Override
	public String toString() {
		return "PokeType [idT=" + idT + ", name=" + name + "]";
	}

	@Override
	public int compareTo(PokeType o) {
		int result = this.getName().compareTo(o.getName()); 
		if (result > 0)
			return 1;
		else if (result == 0)
			return 0;
		return -1;
	}

}
